### PR TITLE
Add concierge cancel confirm page

### PR DESCRIPTION
**Pull request by unDemian**
## Summary
This PR introduces a cancel landing page that will require customer confirmation before actually cancelling a concierge appointment. The confirmation shared component was refactored to be more generic so that multiple buttons are allowed as children components.

**More context 540-hg-gh.**

Succesfull cancellation flow: https://cloudup.com/cjsQ7bd7ItM
Cancellation error flow: https://cloudup.com/cJZZngWUQib

## Testing
1. Book an appointment by following the steps in https://github.com/Automattic/wp-calypso/pull/21364
2. Access the MC happiness/concierge page, under the `Lighthouse Only` tab you should see your appointment
3. Grab the id of your appointment, for now the only way to do that is by inspecting the element https://cloudup.com/c8dX8XZ2kOf and copying the `data-appointment-id` value
4. Access `http://calypso.localhost:3000/me/concierge/<business-site-slug>/<appointmentId>/cancel`
5. Cancel the appointment
6. It should show the confirmation page
7. Refresh the page and try to cancel again
8. An error should be shown

### CODE REVIEW

File: client/me/concierge/cancel/index.js [hunk 1]
```diff
@@ -29,37 +29,64 @@ class ConciergeCancel extends Component {
...
 				return (
-					<Card highlight="error">
-						{ translate( "We couldn't cancel your session, please try again later." ) }
-					</Card>
-				);
+					<Confirmation
+						description={ translate(
+							'You can also reschedule your session. What would you like to?'
+						) }
+						title={ translate( 'Cancel your Concierge session' ) }
```
Comment (a8ci18n) [inline client/me/concierge/cancel/index.js]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Cancelling your Concierge session…' )` *ES Score: 16*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/concierge-cancelling-flow)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/me/concierge/cancel/index.js [hunk 2]
```diff
@@ -29,37 +29,64 @@ class ConciergeCancel extends Component {
...
-			case CONCIERGE_STATUS_CANCELLING_ERROR:
+			default:
 				return (
-					<Card highlight="error">
-						{ translate( "We couldn't cancel your session, please try again later." ) }
-					</Card>
-				);
+					<Confirmation
+						description={ translate(
+							'You can also reschedule your session. What would you like to?'
```
Comment (mattwondra) [inline client/me/concierge/cancel/index.js]: This is missing a word, "do" — should be:

> You can also reschedule your session. What would you like to do?

Comment (unDemian) [reply client/me/concierge/cancel/index.js]: Great catch, updated.

File: client/me/concierge/cancel/index.js [hunk 3]
```diff
@@ -29,37 +29,64 @@ class ConciergeCancel extends Component {
...
 				return (
-					<Card highlight="error">
-						{ translate( "We couldn't cancel your session, please try again later." ) }
-					</Card>
-				);
+					<Confirmation
+						description={ translate(
+							'You can also reschedule your session. What would you like to do?'
+						) }
+						title={ translate( 'Cancel your Concierge session' ) }
```
Comment (a8ci18n) [inline client/me/concierge/cancel/index.js]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Cancelling your Concierge session…' )` *ES Score: 16*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/concierge-cancelling-flow)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**